### PR TITLE
Lua strings sub()

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

Added term entry for Lua Strings function sub() .
File Location: content\lua\concepts\strings\terms\sub\sub.md. Closes #3108 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/strings/terms/sub/sub.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+Title: '.sub()'
+Description: 'Extracts a substring from a given string'
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/sub/sub.md:3]:
```suggestion
Description: 'Extracts a substring from a given string.'
```

File: content/lua/concepts/strings/terms/sub/sub.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`.sub()`** function is used to extract a substring from a given string. The function takes two arguments; the string from which you want to extract a substring and the starting and ending indices of the substring.
+
+## Syntax
+
+```lua
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/sub/sub.md:19]:
```suggestion
```pseudo
```

File: content/lua/concepts/strings/terms/sub/sub.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Here are some examples of how to use `string.sub()` :
+
+```lua
+local originalString = "Hello, World!"
+local subString1 = string.sub(originalString, 1, 5)
+local subString2 = string.sub(originalString, 8)
+
+print(subString1)
+print(subString2)
+
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/sub/sub.md:38]: remove empty line

File: content/lua/concepts/strings/terms/sub/sub.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+## Syntax
+
+```lua
+string.sub(originalString, startIndex, endIndex)
+```
+
+`originalString`: The original string to extract the substring.
+`startIndex`: The index at which the substring should start (inclusive).
+`endIndex` (optional): The index at which the substring should end (exclusive). If `endIndex` is not provided, the substring will extend to the end of the original string.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/sub/sub.md:25]:
```suggestion
- `originalString`: The string from which the substring will be extracted.
- `startIndex`: The index at which the substring should start (inclusive).
- `endIndex` (optional): The index at which the substring should end (exclusive). If `endIndex` is not provided, the substring will extend to the end of the original string.
```

File: content/lua/concepts/strings/terms/sub/sub.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  - 'Game Development'
+Tags:
+  - 'Strings'
+  - 'Functions'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`.sub()`** function is used to extract a substring from a given string. The function takes two arguments; the string from which you want to extract a substring and the starting and ending indices of the substring.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/sub/sub.md:15]:
```suggestion
The **`.sub()`** function is used to extract a substring from a given string. The extracted substring is determined by the integers passed to the function, which specify the starting and ending index.
```

**Review by caupolicandiaz — APPROVED** [review]
@Brag2gr8 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by Brag2gr8** [comment]
@caupolicandiaz I am done with the update

**Review by SSwiniarski — COMMENTED** [review]
@Brag2gr8 Looks good.

**Pull request merged**